.changeset/nervous-parrots-rush.md

```diff
@@ -0,0 +1,47 @@
+---
+'@open-wc/scoped-elements': minor
+---
+
+BREAKING CHANGE: Work without polyfill if possible (and do not auto load polyfill)
+
+ScopedElementsMixin 2.x tried to be as convenient as possible by automatically loading the scoped custom elements registry polyfill.
+This however led to a fatal error whenever you registered any component before ScopedElementsMixin was used.
+
+This especially happens when you import a web component using ScopedElementsMixin into an existing application.
+
+This fundamentally breaks the way we think web components should work. We believe it should be "import and use".
+That is why we decided to treat this change as if it were a security-breaking change - consumers should do the breaking bug fix on their end, but releasing a new major version would not be beneficial, both to us and our consumers.
```

Comment:
```suggestion
The error especially happened when you would import a component applying ScopedElementsMixin into an existing application, fundamentally going against the "import and use" nature of web components.

Therefore, we decided to not load the polyfill inside the mixin, but let the developer (optionally) load it on top of his/her app.
This means that, depending on the app of the developer, the change can be breaking. Two scenarios will be possible:
1. the app doesn't need scoping (in this case the app either continues to work as is, or the fatal error mentioned above will disappear)
2. the app needs scoping, because different versions of the same component are used (in this case, the fatal error mentioned above will disappear and an error asking you to load the polyfill on top of your app will appear )

Only the second case is breaking: the fatal error can be resolved via a small migration path. A breaking bugfix if you will, which should be treated as if it were a security breaking change: consumers do the breaking bug fix on their end, because releasing a new major version would not be beneficial, neither to us nor to our consumers.
```